woocommerce/woocommerce-ios
# Integrating Search Orders Action

### Note:
Includes PR #557 
Includes PR #558 

Please review once the former have been merged back into Develop (and, of course, merge develop back in, the code delta should be ~500 LOC, otherwise it's a review nightmare!).

### Details:
- Integrates the Search Orders UI with the Yosemite Action
- Excludes Orders marked as **exclusiveSearchResult** from the Orders List
- Cached Search results are available in offline mode. Ain't that awesome?

Closes #63

### Testing: Main Functionality
1. Open the Orders Tab and press the top right button (Magnifying Glass)

- [x] Verify the spinner shows up whenever we're in empty state
- [x] Verify the UI effectively retrieves orders that contain the Search Keyword
- [x] Verify Infinite Scroll effectively loads the next page
- [x] Verify the spinner shows up whenever the **next** page is being loaded

### Testing: Offline Mode
1. Open the Orders Tab and press the top right button (Magnifying Glass)
2. Search for a keyword that return a few orders
3. Toggle Airplane Mode
4. Dismiss the Search UI + Reopen it

- [x] Verify that the keyword entered in (2) yields the same results

### Testing: Cache Interference
1. Open the Orders Tab and press the top right button (Magnifying Glass)
2. Enter a keyword that yields an OLD order (that doesn't show up in the top 25 results)
3. Toggle Airplane Mode
4. Dismiss the Search UI
5. Scroll all the way down

- [x] Verify that the orders displayed in Step 2 (that were never cached because of a regular Paging OP)  do not show up, mixed up in the results
- [x] Verify that after disabling Airplane Mode, and scrolling down, such order shows up correctly

### Testing: Fulfillment
1. Open the Orders Tab and press the top right button (Magnifying Glass)
2. Enter any keyword that yields an unfulfilled order
3. Open the first Search Result
4. Fulfill the order

- [x] Verify that the Order Details UI reflects the new Order State
- [x] Verify that the Search Results also reflect the new order State
- [x] Verify that the Orders List is also in sync (dismiss the search UI!)

### Testing: Notes
1. Open the Orders Tab and press the top right button (Magnifying Glass)
2. Enter any keyword that yields any order, and open it

- [x] Verify that adding an Order Note works fine!
- [x] Verify that the Order (when opened via the Orders List) is in sync
